pkg/reconciler/cluster/controller.go

```diff
@@ -126,30 +104,32 @@ func NewController(
 			}
 			return []string{}, nil
 		},
-	}); err != nil {
-		return nil, fmt.Errorf("Failed to add indexer for APIResourceImport: %w", err)
+	}
+
+	// Ensure the indexers are only added if not already present.
+	for indexName := range c.apiresourceImportIndexer.GetIndexers() {
+		delete(indexers, indexName)
+	}
+	if len(indexers) > 0 {
+		if err := c.apiresourceImportIndexer.AddIndexers(indexers); err != nil {
+			return nil, fmt.Errorf("Failed to add indexer for APIResourceImport: %w", err)
+		}
 	}
 
 	return c, nil
 }
 
-type Controller struct {
-	queue                        workqueue.RateLimitingInterface
-	apiExtensionsClient          apiextensionsclient.Interface
-	kcpClient                    kcpclient.Interface
-	clusterIndexer               cache.Indexer
-	apiresourceImportIndexer     cache.Indexer
-	syncChecks                   []cache.InformerSynced
-	syncerImage                  string
-	kubeconfig                   clientcmdapi.Config
-	resourcesToSync              []string
-	syncerMode                   SyncerMode
-	syncerCancelFuncs            map[string]func()
-	apiImporters                 map[string]*APIImporter
-	genericControlPlaneResources []schema.GroupVersionResource
+type ClusterReconciler struct {
```

Comment:
Oh I see why there was an `Impl interface`, to avoid this naming conflict. Could this just be unexported `clusterReconciler` (or `reconciler`, or anything honestly) to avoid that? `config.New` could return the interface?